### ISSUE TITLE
Clarification on the released pretrained checkpoint and encoder branch training

### ISSUE THREAD

**Issue by JJY0329** [weixinxu666/BrainGFM#4]
Thank you for releasing the BrainGFM code and pretrained checkpoint.

I am currently trying to use the released checkpoint for a downstream fMRI study, and I would appreciate some clarification regarding the multi-branch encoder implementation used during pretraining.

In the current `BrainGFM_Gprompt.py`, the relevant part of the encoder forward pass appears to follow this structure:

```python
out = 0

for i in range(len(self.ugformer_layers)):
    h = self.ugformer_layers[i](
        x,
        src_key_padding_mask=padding_mask
    )

    node_h = h[:, 2:, :]
    node_mask = (~padding_mask[:, 2:]).unsqueeze(-1).float()

    node_h = self.lst_gnn[i](
        node_h,
        Adj_block_with_tokens[:, 2:, 2:]
    )

    g = (node_h * node_mask).sum(dim=1) / \
        node_mask.sum(dim=1).clamp(min=1e-6)

    out += self.predictions[i](self.dropouts[i](g))

return g
```

From my understanding, each UGFormer/GNN branch processes the same input `x`, but the function returns only `g` from the final iteration. Therefore, if the pretraining losses are computed from the returned encoder representation, it seems that only the final UGFormer/GNN branch would be connected to those losses.

In addition, although the outputs of `self.predictions[i]` are accumulated in `out`, `out` does not appear to be returned or otherwise used by the pretraining objective.

I may be overlooking some additional logic in the pretraining wrapper, so I wanted to ask:

1. Was the released pretrained checkpoint produced using the current implementation of `BrainGFM_Gprompt.py`?
2. During pretraining, were all four UGFormer/GNN branches expected to receive gradients, or was only the final branch intended to produce the pretrained graph representation?
3. Is `return g` the intended behavior, or were the branches intended to be connected sequentially or combined through `out`?
4. Could you provide the exact pretraining code, configuration, or commit used to produce the released checkpoint?
5. If `BrainGFM_pretrain_remask.py` was used to train the checkpoint, would it be possible to release that implementation as well?

Clarifying this would be very helpful for correctly loading and using the released pretrained weights. If the checkpoint was trained using a different version of the encoder or pretraining code, sharing that version would greatly help with reproducibility.

Thank you for your time and for making the project available.
